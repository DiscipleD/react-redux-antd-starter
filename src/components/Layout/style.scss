.layout{
  height: 100%;
  display: flex;
  flex-flow: column;

  .layout-header{
    height: 64px;
  }

  .layout-body{
    background: #fff;
    flex: 1;
    overflow: hidden;
  }

  .layout-aside{
    width: 224px;
    height: 100%;
    float: left;
  }

  .layout-content{
    height: 100%;
    padding: 0 24px;
    overflow: auto;
  }
}