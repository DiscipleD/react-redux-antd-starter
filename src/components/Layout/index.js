/**
 * Created by jack on 16-9-7.
 */

import React from 'react';

import './style.scss';

class Layout extends React.Component {
	render() {
		const {header, aside, content} = this.props;
		return (
			<div className="layout">
				<header className="layout-header">{header}</header>
				<div className="layout-body">
					<aside className="layout-aside">{aside}</aside>
					<div className="layout-content">{content}</div>
				</div>
			</div>)
	}
}

export default Layout;
