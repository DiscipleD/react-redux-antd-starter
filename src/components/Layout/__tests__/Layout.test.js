/**
 * Created by jack on 16-9-25.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Layout from '../index';

describe('Layout Component', () => {
	it('match snapshot', () => {
		const layout = renderer.create(
			<Layout
				header="header"
				aside="aside"
				content="content"
			/>
		).toJSON();
		expect(layout).toMatchSnapshot();
	});
});
