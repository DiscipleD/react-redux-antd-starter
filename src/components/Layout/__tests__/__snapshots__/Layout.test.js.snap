exports[`Layout Component match snapshot 1`] = `
<div
  className="layout">
  <header
    className="layout-header">
    header
  </header>
  <div
    className="layout-body">
    <aside
      className="layout-aside">
      aside
    </aside>
    <div
      className="layout-content">
      content
    </div>
  </div>
</div>
`;
